build.go

```diff
@@ -108,24 +111,21 @@ func Build(buildManager BuildManager, clock chronos.Clock, logger scribe.Logger)
 	}
 }
 
-func layerFlags(entries []packit.BuildpackPlanEntry) []packit.LayerType {
-	var flags []packit.LayerType
-
+func setLayerFlags(layer packit.Layer, entries []packit.BuildpackPlanEntry) packit.Layer {
 	for _, entry := range entries {
 		launch, ok := entry.Metadata["launch"].(bool)
 		if ok && launch {
-			flags = append(flags, packit.LaunchLayer)
-			flags = append(flags, packit.CacheLayer)
+			layer.Launch = entry.Metadata["launch"] == true
+			layer.Cache = entry.Metadata["launch"] == true
```

Comment:
Why double checking for the same value?
```suggestion
			layer.Launch = true
			layer.Cache = true
```

Reply:
I don't think these values should be hardcoded either. You should be reading them from the metadata.

Reply:
For some reason I thought that if launch is true, we set cache to true too in some instances.

Reply:
@arjun024 @ryanmoran when I switch `layer.Cache = entry.Metadata["build"] == true` to `layer.Cache = entry.Metadata["cache"] == true` the tests fail on [this line](https://github.com/paketo-buildpacks/npm-install/blob/48a26640767ded39517ac7674b36f21ad9be0d8f/integration/caching_test.go#L178).
This is the caching test, and it's failing because the first and second image IDs are no longer equal. Is this something we care about testing?

Reply:
@ryanmoran I think it's ok because by the time line 118 and 119 is hit, we have already checked the value from the metadata on line 116

Reply:
LGTM. @ryanmoran all good for me to merge this?

Reply:
I see. That context was missing from the diff. This seems reasonable.